agent/config/parse.go

```diff
@@ -34,12 +34,6 @@ func parseCheckpoint(dataDir string) BooleanDefaultFalse {
 		if checkPoint.Value == NotSet {
 			checkPoint.Value = ExplicitlyEnabled
 		}
-	} else {
-		// if the directory is not set, default to checkpointing off for
-		// backwards compatibility
-		if checkPoint.Value == NotSet {
-			checkPoint.Value = ExplicitlyDisabled
```

Comment:
https://github.com/aws/amazon-ecs-agent/blob/2874131a5f3d1c0908e0f2ebad62e48b8ff5850f/agent/config/conditional.go#L35-L37
there it is.  So this ends up true to the original intent when `NotSet`